### ISSUE TITLE
OJS 3.1.2 in localized author names the site default locale is required even if the journal does not use it

### ISSUE THREAD

**Issue by ajnyga** [pkp/pkp-lib#5136]
**Describe the bug**
In cases where the site default locale is different from the locale a journal is using, the users are required to fill the Given Name field using the site locale. This means that the names have to be filled in twice. Also the suggest button does not work unless you fill the site locale fields.

Expected behaviour would be to require only the names according to the journal settings and selected languages.

**To Reproduce**
1. Create a new site and set site default locale to fi_FI
2. Create a journal and set default locale to en_US
3. Create a new user account and only fill the en_US name fields

**What application are you using?**
 OJS 3.1.2

**Comment by NateWr** [comment]
Is this a duplicate of https://github.com/pkp/pkp-lib/issues/4818?

**Comment by NateWr** [comment]
I think the problem here is that users are shared across the site. So that might be why a user is thought to require an entry in the site default.

I agree that we shouldn't be asking users for this, and it shouldn't be a requirement for _authors_, but I'm guessing that was the thinking behind it.

**Comment by ajnyga** [comment]
Yes agree that must be the reason. If we have to have all names saved using the site default locale, then this should be automated for journals that do not use the site locale as their default locale.

**Comment by alexxxmendonca** [comment]
Should we even collect translations for _names_? What's the purpose?

What if the system automatically copied over the names to their respective translations while still allowing users to make changes if they want to?

**Comment by ajnyga** [comment]
To make my point more apparent, this is from a journal that only uses English. But when you go to create a new user, the locale field for Finnish (Suomi) is visible and you have to fill that field to save the form. 

<img width="793" alt="Screen Shot 2019-10-10 at 14 07 49" src="https://user-images.githubusercontent.com/16347527/66563884-6a5ab880-eb67-11e9-8925-0445e66cee8f.png">

**Comment by alexxxmendonca** [comment]
I see.

But then, we are treating the exception as a general rule.

**Comment by alexxxmendonca** [comment]
In SciELO's case, we offer OJS in Portuguese, English and Spanish. Each journal can select their own main locale but we can't tell which locale the end user has chosen.

Not to mention that at this time, there are a lot of Portuguese translations missing therefore a lot of people are using the English locale while still having Portuguese as their main locale, causing this problem right here.

**Comment by ajnyga** [comment]
https://github.com/pkp/pkp-lib/issues/4929 is almost a duplicate, but since it mentions the suggest username feature as the main thing, not closing it yet. But it will be solved together with this one.

**Comment by ajnyga** [comment]
Would an acceptable fix here be that if the Context is not using the Site Default Locale, then only the Context locale is required and the Site Default Locale values are automatically filled with the same values in the background? @NateWr @asmecher ?

**Comment by asmecher** [comment]
I'll give this a cautious yes. We do risk some data "poisoning" here, i.e. automatically entering data into another locale when it may not be appropriate, but in the case of names that will be a very rare situation.

**Comment by mannemark** [comment]
Has this issue been addressed in any of the versions since? I believe we have encountered this problem as well as above mentioned [#4929](https://github.com/pkp/pkp-lib/issues/4929) in one of our multilingual journals. Running v 3.3.0.15. Thanks

**Comment by ajnyga** [comment]
I have to check because I just got a new support ticket related to this. Our installation has Finnish as the default.
